cmd/secret-wrapper/main.go

```diff
@@ -202,3 +207,23 @@ func createSecret(client coreclientset.SecretInterface, name, dir string, dry bo
 	}
 	return nil
 }
+
+// uploadKubeconfig will do a best-effort attempt at uploading a kubeconfig
+// file if one does not exist at the time we start running but one does get
+// created while executing the command
+func uploadKubeconfig(client coreclientset.SecretInterface, name, dir string, dry bool) {
+	if _, err := os.Stat(path.Join(dir, "kubeconfig")); err == nil {
+		// kubeconfig already exists, no need to do anything
+		return
+	}
```

Comment:
This could checked before launching the go routine.

Reply:
Yes, but as written it keeps things self-contained in this method - any downside to starting a Goroutine that exits quickly?